### PR TITLE
added API routes

### PR DESCRIPTION
**Pull request by cpietsch**
This adds 3 API routes, mirroring the behaviour of the extension and exposing those endpoints to the automatic1111 API.
In response to [Feature Request] : Ext through API #52 and https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/11190

### CODE REVIEW

**Review by pharmapsychotic — COMMENTED** [review]
Looks awesome, thanks so much for the PR!! Does it work with a fresh install of the web ui? I can give it a test after work tonight but not sure just from looking at code if there will be a problem with the pydantic and fastapi dependencies.

File: scripts/clip_interrogator_ext.py [hunk 1]
```diff
@@ -11,7 +12,12 @@
 
 from modules import devices, lowvram, script_callbacks, shared
 
-__version__ = '0.1.5'
+from pydantic import BaseModel, Field
+from fastapi import FastAPI
```
[the remark this answers is not in the document]
Comment (cpietsch) [reply scripts/clip_interrogator_ext.py:16]: Both should be installed, probably through gradio.
See https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/modules/api/models.py and https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/modules/api/api.py

[the remark this answers is not in the document]
Comment (pharmapsychotic) [reply scripts/clip_interrogator_ext.py:16]: Ah yep looks like gradio brings 'em in https://github.com/gradio-app/gradio/blob/main/requirements.txt

**Review by pharmapsychotic — APPROVED** [review]
Fantastic addition! 👏

**Comment by cpietsch** [comment]
Thanks! Also thank you for this awesome extension!

File: scripts/clip_interrogator_ext.py [hunk 2]
```diff
@@ -296,4 +302,73 @@ def add_tab():
...
+    async def get_prompt(analysereq: InterrogatorPromptRequest):
+        image_b64 = analysereq.image
+        if image_b64 is None:
+            raise HTTPException(status_code=404, detail="Image not found")
+
+        img = decode_base64_to_image(image_b64)
+        prompt = image_to_prompt(img, analysereq.mode, analysereq.clip_model_name)
+        return {"prompt": prompt}
+
+    @app.post("/interrogator/analyse")
```
[the remark this answers is not in the document]
Comment (cpietsch) [reply scripts/clip_interrogator_ext.py]: Done: db4340b30f85ba0c74ad53bf564ec4dd785de250

**Comment by MeinDeutschkurs** [comment]
WOOOHOOO!!! ❤️ Thank you so much for that awesome addition to that awesome extension!! ❤️

**Pull request merged**